### PR TITLE
Tc 5008

### PR DESCRIPTION
**Pull request by josethirdchannel**
https://thirdchannel.atlassian.net/browse/TC-5008
This adds an extra check at submission time to make sure a mentioned user is still contained in the final text of the comment

### CODE REVIEW

File: js/apps/thirdchannel/views/comments/new_comment.js [hunk 1]
```diff
@@ -173,10 +174,19 @@ define(function(require) {
             });
 
             if (comment.isValid()) {
+                var mentions = [];
+                var possibleMentions = this.$('.new-comment-field').data('mentions');
+                var text = this.$('.new-comment-field').text();
+                Object.keys(possibleMentions).forEach(function (key) {
+                    if (text.includes(mentions[key])) {
```
Comment (eliottwiener-thirdchannel) [inline js/apps/thirdchannel/views/comments/new_comment.js]: I think you meant to use `possibleMentions[key]` here

File: js/apps/thirdchannel/views/comments/new_comment.js [hunk 2]
```diff
@@ -46,7 +46,8 @@ define(function(require) {
                     var splitLabel = ui.item.originalText.split("\t");
```
Comment (eliottwiener-thirdchannel) [inline js/apps/thirdchannel/views/comments/new_comment.js]: It's not immediately obvious what is going on in this `displayItem` function. I had to run through it with some log statements to figure out what was going on. Some comments might make it easier to understand.

Comment (josethirdchannel) [inline js/apps/thirdchannel/views/comments/new_comment.js]: 👍 agreed

**Comment by josethirdchannel** [comment]
tested by disabling Defenestration and checking the messages sent on the notification topic. It looks like they're still being created as expected but now only include the recipientUUIDs of someone still mentioned in the text on submission

**Pull request merged**